website/src/components/edit-word-audio/contributor.tsx

```diff
@@ -121,35 +154,70 @@ export function ContributorEditWordAudio(p: {
   )
 
   return (
-    <div>
+    <>
       {[curatedAudioContent, userAudioContent, otherUserAudioContent]
         .filter((content) => content !== false)
         .map((content, idx) => (
-          <>
+          <Fragment key={idx}>
             {idx > 0 && <hr />}
             {content}
-          </>
+          </Fragment>
         ))}
+    </>
+  )
+}
+
+function ContributeAudioSection(p: {
+  word: Dailp.FormFieldsFragment
+}): ReactElement {
+  const [currentTab, setCurrentTab] = useState<"upload" | "record">()
+  const [selectedFile, setSelectedFile] = useState<File>()
+  const [uploadAudio, uploadState, clearUploadError] = useAudioUpload(p.word.id)
+
+  function onFileChanged(event: ChangeEvent<HTMLInputElement>) {
+    if (!event.currentTarget.files) return
+
+    const [file] = event.currentTarget.files
+    if (!file) return
+    console.log("New file selected", file)
+    setSelectedFile(file)
+    setCurrentTab("upload")
+  }
+
+  async function uploadAudioAndReset(data: Blob) {
+    const success = await uploadAudio(data)
+    if (success) {
+      setCurrentTab(undefined)
+      setSelectedFile(undefined)
+    }
+  }
+
+  return (
+    <>
       <VisuallyHidden>
         <input type="file" id="file-upload" onChange={onFileChanged} />
       </VisuallyHidden>
-      <div
+      <form
         style={{
           display: "flex",
           textAlign: "center",
           justifyContent: "space-between",
         }}
       >
-        <label htmlFor="file-upload" style={{ display: "inline-block" }}>
+        <div style={{ display: "inline-block" }}>
           <IconTextButton
             icon={<MdUploadFile />}
+            // @ts-ignore -- I couldn't find a way to fix how this is typed
+            // this is part of React's ugly insides
+            as="label"
+            htmlFor="file-upload"
             className={
               currentTab === "upload" ? subtleButtonActive : subtleButton
             }
           >
             Upload audio
           </IconTextButton>
-        </label>
+        </div>
```

Comment:
There was some bad interaction happening between the button and label elements here. This seems to have resolved the issue, kept the tab order correct, and kept cursor changes correct as well.